### PR TITLE
[Production checks] Gdrive GC

### PR DESCRIPTION
**Pull request by lasryaric**
Adding production checks for Gdrive.

# Deployment steps:
 Add the following 3 env vars to front:
```
CORE_DATABASE_URI_RO
FRONT_DATABASE_URI_RO
CONNECTORS_DATABASE_URI_RO 
```
- Deploy front
- Start the cron based workflow:
```
./front/production_check> npx tsx ./admin/cli.ts start
```

Check if the workflow is properly running on Temporal Cloud.

## Setup Datadog Alerts
- One alerts that triggers if the log `Production check failed` is emitted
- One alerts that trigger if the log `Done running all checks` is not trigger within a 90 minutes time frame.

### CODE REVIEW

**Comment by spolu** [comment]
We also need CORE_DATABASE_URI and CONNECTORS_DATABASE_URI to be added to front so that we have access to all SQL moving forward

**Comment by lasryaric** [comment]
> We also need CORE_DATABASE_URI and CONNECTORS_DATABASE_URI to be added to front so that we have access to all SQL moving forward

You mean on top of the RO version?
```
CORE_DATABASE_URI_RO
FRONT_DATABASE_URI_RO
CONNECTORS_DATABASE_URI_RO 
```

File: front/production_checks/checks/managed_data_sources_gc.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    documents: CoreDSDocument[],
+    connectorsSequelize: Sequelize
+  ): Promise<Result<void, unknown>> => {
+    const results: CoreDSDocument[] = [];
+    do {
+      const slice = documents.splice(0, 100);
+      const dustFileIds = slice.map((d) => d.document_id);
+
+      const documentsData = await connectorsSequelize.query(
+        'SELECT id as "googleDriveFileId", "dustFileId" FROM google_drive_files WHERE "dustFileId" IN (:documentIds) and "connectorId" = :connectorId',
```
Comment (spolu) [inline front/production_checks/checks/managed_data_sources_gc.ts:156]: I think we should stick with the flow that was used in the migration

Get the files create a hash, get all core files (we can limit to document_id to make it more lightweight), check them all

File: front/production_checks/checks/managed_data_sources_gc.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  slack: async (
+    connectorId: ModelId,
+    documents: CoreDSDocument[],
+    connectorsSequelize: Sequelize
+  ): Promise<Result<void, unknown>> => {
+    const missingChannels: string[] = [];
+    const coreChannel = new Set(documents.map((d) => d.parents[0]));
+    for (const channelId of coreChannel) {
+      const selectedChannel: { id: number; permission: string }[] =
+        await connectorsSequelize.query(
```
Comment (spolu) [inline front/production_checks/checks/managed_data_sources_gc.ts:123]: How is this independent from the permission flag?

**Review by spolu — COMMENTED** [review]
The code is hard to read TBH

Let's make things simple. Separate in two checks one for Slack one for Gdrive

For GDrive let's stick with the migration flow that is easy to understand.

Take all connector file ids and create a hash (that's fine that holds in memory)

Then take all core document_id and check them.

No chunking, it's easier to understand and can be read more easily

The fact that you try to do both at the same time makes it hard to understand and read. We need DEAD simple here otherwise we'll not maintain it.

File: front/production_checks/checks/managed_data_sources_gc.ts (hunk 2 above)
Comment (lasryaric) [reply front/production_checks/checks/managed_data_sources_gc.ts:123]: Will handle Slack in a different PR. This was a mistake from my side.

File: front/production_checks/checks/managed_data_source_gdrive_gc.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const coreDocumentIds = coreDocuments.map((d) => d.document_id);
+    const connectorDocumentIds = new Set(
+      connectorDocuments.map((d) => d.coreDocumentId)
+    );
+    const missingDocuments = coreDocumentIds.filter(
+      (coreId) => !connectorDocumentIds.has(coreId)
+    );
+    if (missingDocuments.length > 0) {
+      reportFailure(
+        { missingDocuments },
```
Comment (spolu) [inline front/production_checks/checks/managed_data_source_gdrive_gc.ts:56]: missingDocument is misnamed they're not missing they're notDeleted

File: front/production_checks/checks/managed_data_source_gdrive_gc.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  reportSuccess,
+  reportFailure
+) => {
+  const connectorsSequelize = new Sequelize(
+    CONNECTORS_DATABASE_READ_REPLICA_URI as string,
+    {
+      logging: false,
+    }
+  );
+  const GdriveDataSources = await DataSource.findAll({
```
Comment (spolu) [inline front/production_checks/checks/managed_data_source_gdrive_gc.ts:20]: Let's not use specific models so that we're not tied to `front` in any way

**Pull request merged**